src/expressions/match-expr.md

```diff
@@ -75,17 +76,21 @@ let message = match x {
 assert_eq!(message, "a few");
 ```
 
-Please notice that the `2..=9` is a [Range Pattern], not a [Range Expression]
-and, thus, only those types of ranges supported by range patterns can be used
-in match arms.
+> Note: The `2..=9` is a [Range Pattern], not a [Range Expression] and, thus,
+> only those types of ranges supported by range patterns can be used in match
+> arms.
+
+Every binding in each `|` separated pattern must appear in all of the patterns
+in the arm. Every binding of the same name must have the same type, and have
```

Comment:
This is a bit vague since it doesn't say whether "same type" accounts for subtyping or not, but it's serviceable until we have a more formal semantics.